db/db_impl/db_impl_write.cc

```diff
@@ -1432,10 +1432,23 @@ WriteBatch* DBImpl::MergeBatch(const WriteThread::WriteGroup& write_group,
 IOStatus DBImpl::WriteToWAL(const WriteBatch& merged_batch,
                             log::Writer* log_writer, uint64_t* log_used,
                             uint64_t* log_size,
-                            LogFileNumberSize& log_file_number_size) {
+                            LogFileNumberSize& log_file_number_size,
+                            int caller_id) {
   assert(log_size != nullptr);
-
+  if (log_writer->file()->GetFileSize() == 0) {
+    ROCKS_LOG_INFO(immutable_db_options_.info_log,
+                   "Start writing to WAL: [%" PRIu64 " ]",
+                   log_writer->get_log_number());
+  }
+  if (log_writer->get_log_number() != logs_.back().number) {
```

Comment:
@tonyxuqqi : At the beginning of the write, writer will reference the last WAL only. The race condition happens, while the write is ongoing. We should check this condition after the write.